doc/sphinx/proof-engine/tactics.rst

```diff
@@ -105,6 +108,277 @@ bindings_list`` where ``bindings_list`` may be of two different forms:
   .. exn:: Not the right number of missing arguments.
      :undocumented:
 
+.. _intropatterns:
+
+Intro patterns
+~~~~~~~~~~~~~~
+
+Intro patterns let you specify the name to assign to variables and hypotheses
+introduced by tactics.  They also let you split an introduced hypothesis into
+multiple hypotheses or subgoals.  Common tactics that accept intro patterns
+include :tacn:`apply`, :tacn:`intros` and :tacn:`destruct`.
+
+.. productionlist::        coq
+   ne_intropatterns        : `intropattern` ... `intropattern`
+   intropatterns           : `intropattern` ... `intropattern`
+                           : `empty`
+   empty                   :
```

Comment:
I like a lot this way you found to make the empty case explicit, and it is also clear why this is needed. However, I would avoid repeating the ``` `intropattern` ... `intropattern` ``` part. So I suggest changing the above to:

```
    intropatterns           : `ne_intropatterns`
                            : `empty`
    ne_intropatterns        : `intropattern` ... `intropattern`
    empty                   :